configuration_version.go

```diff
@@ -11,12 +11,23 @@ import (
 	slug "github.com/hashicorp/go-slug"
 )
 
-// ConfigurationVersions handles communication with the configuration version
-// related methods of the Terraform Enterprise API.
+// Compile-time proof of interface implementation.
+var _ ConfigurationVersions = (*configurationVersions)(nil)
+
+// ConfigurationVersions describes all the configuration version related
+// methods that the Terraform Enterprise API supports.
 //
 // TFE API docs:
 // https://www.terraform.io/docs/enterprise/api/configuration-versions.html
-type ConfigurationVersions struct {
+type ConfigurationVersions interface {
+	List(context.Context, string, ConfigurationVersionListOptions) ([]*ConfigurationVersion, error)
+	Create(context.Context, string, ConfigurationVersionCreateOptions) (*ConfigurationVersion, error)
+	Read(context.Context, string) (*ConfigurationVersion, error)
+	Upload(context.Context, string, string) error
```

Comment:
I think defining the interface without naming the args will make the godoc much less useful. What are the two strings this method wants? It's hard to tell without the names.

We are also losing all of the method godocs, because they now only exist on private structs. I think it's valuable to preserve that in the interface so that users can see the godoc properly.

Reply:
Good points, will update that...